src/cockpit/beiboot.py

```diff
@@ -44,6 +44,12 @@
 
 
 def ensure_ferny_askpass() -> Path:
+    # packaged case
+    askpass = Path(get_libexecdir(), 'cockpit-askpass')
```

Comment:
I almost feel like we should do `patch_libexecdir('${libexecdir}/cockpit-askpass')` to hit the special case logic in that function for falling back to the path.  In that case, I can't imagine an installed scenario where we would *ever* have to create the temporary version as we do below, which would effectively turn that into dead code...

The place this would make a difference, of course, is when running out of tree.  Then there's certainly no `cockpit-askpass`....

Let's discuss this further.

Reply:
So patch_libexecdir() would make a difference when installing the wheel, indeed. Not sure if that should be a supported use case, but it doesn't hurt -- this isn't performance critical. So I'm happy to do that.

> The place this would make a difference, of course, is when running out of tree

That I don't understand -- then it won't be in libexecdir nor in $PATH, so it is meant to go with "write temp file".

Sure, let's talk  about it this morning. Thanks!

Reply:
Result of conversation: this complicated code exists because the flatpak needs the askpass executable to exist on the *host* system, which means we need to copy it to the homedir.

Let's take the check for `/.flatpak-info` in `do_connect_transport()` and use that to decide if we need the copy or if we should just consult `patch_libexecdir()`.

Reply:
Notes from call: We need the xdg-cache case exactly for the flatpak (libexecdir is in the flatpak container, but askpass needs to run on the host, with ssh) → add a comment, check for flatpak, and call either ensure_ferny_askpass() or patch_libexecdir().

Reply:
aside: it's disturbing that this PR passed testing.  We probably should figure out a way to add a testcase for the flatpak that involves an authentication interaction.

Reply:
FTR: the flatpak works, as it's not using the libexecdir path, andthe flatpak's HOME is also visible from the host:
```
cockpit.packages-WARNING: Could not detect libexecdir
cockpit.beiboot-DEBUG: Checking if /home/martin/.var/app/org.cockpit_project.CockpitClient/cache/cockpit-client-askpass exists...
cockpit.beiboot-DEBUG: Launching command: cmd=('flatpak-spawn', '--host', '--env=SSH_ASKPASS=/home/martin/.var/app/org.cockpit_project.CockpitClient/cache/cockpit-client-askpass', '--env=DISPLAY=x', '--env=SSH_ASKPASS_REQUIRE=force', 'ssh', '-p', '2201', 'admin@127.0.0.2', "python3 -ic '# cockpit-bridge'") env=()

```

Reply:
The reason this works is by accident: it's only because our libexecdir-finding code is currently not smart enough to consider `/app/libexecdir`, causing it to fail to find `cockpit-askpass` and allowing this to succeed by accident.

We can deal with that whole mess in a different PR, possibly #19264.